File: app/src/main/java/com/example/alarm/room/SettingsDao.java
package com.example.alarm.room;

import androidx.room.Dao;
import androidx.room.Query;
import androidx.room.Update;

import kotlinx.coroutines.flow.Flow;

@Dao
public interface SettingsDao {
    @Query("SELECT * FROM settings")
    SettingsDbEntity getSettings();

    @Update
    void updateSettings(SettingsDbEntity settingsDbEntity);
}

File: app/src/main/java/com/example/alarm/room/SettingsDao.kt
package com.example.alarm.room

import androidx.room.Dao
import androidx.room.Query
import androidx.room.Update
import kotlinx.coroutines.flow.Flow

@Dao
interface SettingsDao {
    @Query("SELECT * FROM settings")
    suspend fun getSettings(): SettingsDbEntity

    @Update(entity = SettingsDbEntity::class)
    suspend fun updateSettings(settingsDbEntity: SettingsDbEntity)
}